runelite-client/src/main/java/net/runelite/client/plugins/worldmap/WorldMapPlugin.java

```diff
@@ -271,4 +287,57 @@ private void updateShownIcons()
 			}).map(TeleportPoint::new)
 			.forEach(worldMapPointManager::add);
 	}
+
+	private void updateQuestStartPointIcons()
+	{
+		if (!config.questStartTooltips()) return;
+
+		worldMapPointManager.removeIf(QuestStartPoint.class::isInstance);
+		Arrays.stream(QuestStartLocation.values())
+			.map(value -> createQuestStartPoint(value))
+			.forEach(worldMapPointManager::add);
+	}
+
+	private QuestStartPoint createQuestStartPoint(QuestStartLocation data)
+	{
+		Quest[] quests = data.getQuests();
+
+		// Get first uncompleted quest. Else, return the last quest.
+		Quest quest = null;
+		for (int i = 0; i < quests.length; i++)
+		{
+			if (quests[i].getState(client) != QuestState.FINISHED)
+			{
+				quest = quests[i];
+				break;
+			}
+		}
+		if (quest == null)
+		{
+			quest = quests[quests.length - 1];
+		}
+
+		BufferedImage icon = BLANK_ICON;
+		String tooltip = "";
+		if (quest != null)
+		{
+			tooltip = quest.getName();
+			switch (quest.getState(client))
```

Comment:
This call is required to be on the client thread, I think it will assert fail if you run your client with `-ea`. The config change events can come in from other threads.

Reply:
Ah, interesting. I'll see if it fails as expected later this evening. Hopefully I can find a solution by studying how other plugins access the client.

Reply:
I tried changing the config during play, but no crash. I did notice that if you have the world map open, and you toggle the WorldMapPlugin, the quest icons won't come back until you re-open the world map. That's because the icons are not done in `startUp`.

Attempting to access client in `startUp` does crash though:

<details>
<summary>expand</summary>

```
2019-06-11 22:00:36 [main] WARN  n.r.client.plugins.PluginManager - Unable to start plugin WorldMapPlugin. {}
net.runelite.client.plugins.PluginInstantiationException: java.lang.reflect.InvocationTargetException
	at net.runelite.client.plugins.PluginManager.startPlugin(PluginManager.java:342)
	at net.runelite.client.plugins.PluginManager.startCorePlugins(PluginManager.java:207)
	at net.runelite.client.RuneLite.start(RuneLite.java:296)
	at net.runelite.client.RuneLite.main(RuneLite.java:221)
Caused by: java.lang.reflect.InvocationTargetException: null
	at java.desktop/java.awt.EventQueue.invokeAndWait(EventQueue.java:1367)
	at java.desktop/java.awt.EventQueue.invokeAndWait(EventQueue.java:1342)
	at java.desktop/javax.swing.SwingUtilities.invokeAndWait(SwingUtilities.java:1480)
	at net.runelite.client.plugins.PluginManager.startPlugin(PluginManager.java:314)
	... 3 common frames omitted
Caused by: java.lang.AssertionError: null
	at client.runScript(client.java:41459)
	at net.runelite.api.Quest.getState(Quest.java:200)
	at net.runelite.client.plugins.worldmap.WorldMapPlugin.createQuestStartPoint(WorldMapPlugin.java:310)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:497)
	at net.runelite.client.plugins.worldmap.WorldMapPlugin.updateQuestStartPointIcons(WorldMapPlugin.java:299)
	at net.runelite.client.plugins.worldmap.WorldMapPlugin.startUp(WorldMapPlugin.java:129)
	at net.runelite.client.plugins.PluginManager.lambda$startPlugin$3(PluginManager.java:318)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:303)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:770)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:715)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:740)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

</details>

Reply:
you need to invoke `Quest::getState` on the client thread via `ClientThread::invoke` because it calls `Client::getVar`, which is not threadsafe.

Reply:
Alrighty, I figured out the threading constraints. Was able to remove the on widget loaded too.

Reply:
This is still on the wrong thread. If you run the client with JVM option `-ea` and toggle the quest start tooltips config setting you should be able to see it.

Reply:
Ah, I see it now. I was only toggling the entire plugin. Done.